cmd/ndm_daemonset/probe/usedbyprobe.go

```diff
@@ -200,13 +200,14 @@ func getBlockDeviceZFSPartition(bd blockdevice.BlockDevice) (string, bool) {
 // isBlockDeviceInUseByKernel tries to open the device exclusively to check if the device is
 // being held by some process. eg: If kernel zfs uses the disk, the open will fail
 func isBlockDeviceInUseByKernel(path string) (bool, error) {
-	_, err := os.OpenFile(path, os.O_EXCL, 0444)
+	f, err := os.OpenFile(path, os.O_EXCL, 0444)
 
 	if errors.Is(err, syscall.EBUSY) {
 		return true, nil
 	}
 	if err != nil {
 		return false, err
 	}
+	defer f.Close()
```

Comment:
Since we are not doing any activity after opening the file(ex: re-opening the file with the same flags which leads to point to the same fd) it is assumed the `Close` func will not return any error.

Reply:
yes @mittachaitu . We are just opening the file and if open succeeds, we close the file.